### PR TITLE
On password reset, check that new password is different

### PR DESCRIPTION
**Pull request by fisx**

### CODE REVIEW

**Comment by fisx** [comment]
TODO: check for `PUT /password`!

File: services/brig/src/Brig/API/Error.hs [hunk 1]
```diff
@@ -211,6 +212,9 @@ invalidEmail = Wai.Error status400 "invalid-email" "Invalid e-mail address."
 invalidPwResetKey :: Wai.Error
 invalidPwResetKey = Wai.Error status400 "invalid-key" "Invalid email or mobile number for password reset."
 
+newPasswordMustDiffer :: Wai.Error
+newPasswordMustDiffer = Wai.Error status400 "password-must-differ" "For password reset, new and old password must be different."
```
Comment (tiago-loureiro) [inline services/brig/src/Brig/API/Error.hs]: Small nitpick; I'd turn this into a 409

File: services/brig/src/Brig/API/User.hs [hunk 2]
```diff
@@ -606,19 +606,21 @@ mkActivationKey (ActivatePhone p) = do
...
     let newpw = cpNewPassword cp
     case (currpw, cpOldPassword cp) of
-        (Nothing,        _) -> lift $ Data.updatePassword u newpw
+        (Nothing,        _) -> lift $ Data.updatePassword uid newpw
         (Just  _,  Nothing) -> throwE InvalidCurrentPassword
         (Just pw, Just pw') -> do
             unless (verifyPassword pw' pw) $
                 throwE InvalidCurrentPassword
-            lift $ Data.updatePassword u newpw >> revokeAllCookies u
+            when (verifyPassword newpw pw) $
```
Comment (ChrisPenner) [inline services/brig/src/Brig/API/User.hs]: Would it make sense to make `checkNewIsDifferent` top-level and call it here? Maybe pass in the type of error to call? Or is it small enough to not matter?

File: services/brig/src/Brig/Provider/API.hs [hunk 3]
```diff
@@ -398,12 +398,15 @@ beginPasswordReset req = do
 
 completePasswordReset :: JsonRequest CompletePasswordReset -> Handler Response
 completePasswordReset req = do
-    CompletePasswordReset key val pwd <- parseJsonBody req
-    c <- Code.verify key Code.PasswordReset val >>= maybeInvalidCode
-    case Code.codeAccount c of
+    CompletePasswordReset key val newpwd <- parseJsonBody req
```
Comment (ChrisPenner) [inline services/brig/src/Brig/Provider/API.hs]: Thanks for clarifying names :)

File: services/brig/src/Brig/API/User.hs (hunk 2 above)
Comment (fisx) [reply services/brig/src/Brig/API/User.hs]: I'd say it's borderline.  I will leave it as is this time and try to lean more towards smaller functions in the future.

**Pull request merged**